### ISSUE TITLE
Decode unix timestamps into time.Time just like the `UnixTime` option for Marshal?

### ISSUE THREAD

**Issue by takanuva15** [wI2L/jettison#14]
Hi, this library has a great option `UnixTime` that "Encode[s] time.Time values as JSON numbers representing Unix timestamps".

In my case, I have to deal with json that sends time values as integer unix timestamps and unmarshal it into structs using time.Time fields. The default encoding/json just throws an error when it sees the unix timestamps, and our team's codebase is too large to replace every time.Time field with a custom type just for the sake of these int timestamps. 

I'm aware this library is for marshalling only, but do you know if there's a way we can somehow apply this behavior to work for unmarshalling as well, either through this library or another? (I looked at goccy/go-json but I don't see an option for customizing unmarshal for time.Time)

**Comment by wI2L** [comment]
@takanuva15
This library does only encoding, and I have no plan to add unmarshaling support.
I'm not aware of any other lib that support `time.Time` values like this one. The recommended approach would always be to implement a custom type with a specific `json.Unmarshaler` interface if you use a lib (either official or another one) that implement the same API as `encoding/json`.

**Comment by takanuva15** [comment]
Ok after spending hours of looking at Golang code, I finally figured out a way to customize the decoding of epoch milli timestamps into `time.Time` fields in my own struct without needing to make a dummy wrapper type.

Note: This does require the new experimental [json](https://github.com/go-json-experiment/json) package being developed for Golang, which will eventually be converted to `encoding/json/v2` once it gets out of development phase.

`go get github.com/go-json-experiment/json`

Define your struct like so:
```go
type Author struct {
	CreatedAt             *time.Time            `json:"createdAt,omitempty,omitzero,format:unixmilli"`
}
```

And then unmarshal json into the Author struct using the above package:
```go
import (
	"fmt"
	"time"

	"github.com/go-json-experiment/json/v1"
)

func main() {
	jsonStr := `{"createdAt":1734558308666}`
	a := Author{}
	_ = json.Unmarshal([]byte(jsonStr), &a)
	res, _ := json.Marshal(a)
	fmt.Println(string(res))
}
```

Output: `{"createdAt":1734558308666}`

---

EDIT: As a bonus, I got this to work in gin as well for serializing responses using the new json package. See [this comment](https://github.com/gin-gonic/gin/issues/4112#issuecomment-2552369443) for details
